hummingbot/connector/gateway/clob/gateway_sol_clob.py

```diff
@@ -0,0 +1,1252 @@
+            pass
+        else:
+            if not new_nonce:
+                resp_json: Dict[str, Any] = await self._get_gateway_instance().get_evm_nonce(
+                    self.chain, self.network, self.address
+                )
+                new_nonce: int = resp_json.get("nonce")
+
+        self._nonce = new_nonce
+
+    async def _status_polling_loop(self):
+        await self.update_balances(on_interval=False)
+        while True:
+            try:
+                self._poll_notifier = asyncio.Event()
+                await self._poll_notifier.wait()
+                await safe_gather(
+                    self.update_balances(on_interval=True),
+                    self.update_canceling_transactions(self.canceling_orders),
+                    self.update_token_approval_status(self.approval_orders),
+                    self.update_order_status(self.amm_orders)
+                )
+                self._last_poll_timestamp = self.current_timestamp
+            except asyncio.CancelledError:
+                raise
+            except Exception as e:
+                self.logger().error(str(e), exc_info=True)
+
+    async def update_balances(self, on_interval=False):
+        """
+        Calls Eth API to update total and available balances.
+        """
+        if self._native_currency is None:
+            await self.get_chain_info()
+        last_tick = self._last_balance_poll_timestamp
+        current_tick = self.current_timestamp
+        if not on_interval or (current_tick - last_tick) > constant.UPDATE_BALANCE_INTERVAL:
+            self._last_balance_poll_timestamp = current_tick
+            local_asset_names = set(self._account_balances.keys())
+            remote_asset_names = set()
+            resp_json: Dict[str, Any] = await self._get_gateway_instance().get_balances(
+                self.chain, self.network, self.address, list(self._tokens)
+            )
+            for token, bal in resp_json["balances"].items():
+                self._account_available_balances[token] = Decimal(str(bal))
+                self._account_balances[token] = Decimal(str(bal))
+                remote_asset_names.add(token)
+
+            asset_names_to_remove = local_asset_names.difference(remote_asset_names)
+            for asset_name in asset_names_to_remove:
+                del self._account_available_balances[asset_name]
+                del self._account_balances[asset_name]
+
+            self._in_flight_orders_snapshot = {k: copy.copy(v) for k, v in self._order_tracker.all_orders.items()}
+            self._in_flight_orders_snapshot_timestamp = self.current_timestamp
+
+    async def _update_balances(self):
+        """
+        This is called by UserBalances.
+        """
+        await self.update_balances()
+
+    async def cancel_all(self, timeout_seconds: float) -> List[CancellationResult]:
+        """
+        This is intentionally not awaited because cancellation is expensive on blockchains. It's not worth it for
```

Comment:
> Update this text since the orders are being canceled on Solana.

Sorry I didn't understand what you mean. Do you mean to add something like `on Solana blockchains` there?

Reply:
> However, thinking about this in the context of CLOB orders, it probably makes sense that you want to cancel them so they are not executed when you are not running a strategy.

Yes, I agree, the only thing is that if the cancelation fails, the user will not know. But since GatewayEVMAMM was not canceling anything, I think this behavior may be acceptable here.